src/main/webapp/app/assessment/assessment-locks/assessment-locks.component.html

```diff
@@ -0,0 +1,192 @@
+                    </td>
+                    <td>{{ 'artemisApp.tutorExerciseDashboard.languages.' + (textSubmission.language || 'UNKNOWN') | translate }}</td>
+                    <td>
+                        <span *jhiHasAnyAuthority="['ROLE_ADMIN', 'ROLE_INSTRUCTOR', 'ROLE_TA']">
+                            <button
+                                *ngIf="textSubmission && textSubmission.result"
+                                [routerLink]="['/course-management', courseId, 'text-exercises', textSubmission.exerciseId, 'submissions', textSubmission.id, 'assessment']"
+                                class="btn btn-outline-secondary btn-sm mb-1"
+                            >
+                                <fa-icon [icon]="'folder-open'" [fixedWidth]="true"></fa-icon>&nbsp;{{ 'artemisApp.assessment.dashboard.actions.open' | translate }}
+                            </button>
+                        </span>
+                        <span>
+                            <button
+                                *ngIf="textSubmission && textSubmission.result && !textSubmission.result.completionDate"
+                                (click)="cancelTextAssessment(textSubmission)"
+                                class="btn btn-outline-secondary btn-sm mb-1"
+                            >
+                                <fa-icon [icon]="'ban'" [fixedWidth]="true"></fa-icon>&nbsp;{{ 'artemisApp.assessment.dashboard.actions.cancel' | translate }}
+                            </button>
+                        </span>
+                    </td>
+                </tr>
+            </tbody>
+        </table>
+    </div>
+</div>
+
+<div *ngIf="fileUploadSubmissions.length > 0">
+    <h4 style="margin-top: 10px;">
+        {{ 'artemisApp.assessment.locks.fileUploadSubmissions' | translate }}
+    </h4>
+    <div class="table-responsive">
+        <table class="table table-striped exercise-table">
+            <thead>
+                <tr>
+                    <th>#</th>
+                    <th>Id</th>
+                    <th jhiTranslate="artemisApp.assessment.locks.exercise">Exercise</th>
+                    <th jhiTranslate="artemisApp.assessment.dashboard.columns.submissionDate">Submission date</th>
+                    <th jhiTranslate="artemisApp.assessment.dashboard.columns.submissionCount">Submission count</th>
+                    <th jhiTranslate="artemisApp.assessment.dashboard.columns.score">Score</th>
+                    <th jhiTranslate="artemisApp.assessment.dashboard.columns.action">Action</th>
+                </tr>
+            </thead>
+            <tbody>
+                <tr *ngFor="let fileUploadSubmission of fileUploadSubmissions; let i = index">
+                    <td>{{ i + 1 }}</td>
+                    <td>{{ fileUploadSubmission.id }}</td>
+                    <td>{{ getExerciseName(fileUploadSubmission.exerciseId) }}</td>
```

Comment:
if wired up correctly, this could be 
```suggestion
                    <td>{{ fileUploadSubmission.participation.exercise.title }}</td>
```

Reply:
Refactored